### PR TITLE
✨ Implement blank handlers

### PR DESCRIPTION
**Pull request by ElRaptorus**
**Changes:**

1. Add FlowNodeHandler for `EmptyActivity`.
    - This activity is suspendable, interruptible and can be handled just like a ManualTask.
2. Add FlowNodeHandler for `EmptyIntermediateEvent`
    - This handler is not suspendable and just passes on the token to the next FlowNode.
    - This handler is used as a fallback for the `IntermediateEventHandler` factories.
2. Restructure the Model-Parser:
    - Split up the `Activity` parser into smaller parsers, each dedicated to a single type of activity.
    - Group the parsers into namespaces to make their UseCase clearer.
    - Nothing was changed in terms of functionality, so the parser works just as it did before.
3. Add missing Promise-override to Unit-Tests. 

**Issues:**

Part of https://github.com/process-engine/process_engine_runtime/issues/270

PR: #253

## How can others test the changes?

Try running a ProcessModel that contains Activities without any kind of definition attached to it.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

### CODE REVIEW

File: src/model/parser/flow_node_parsers/activity_parsers/user_task_parser.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+import {getModelPropertyAsArray} from '../../../type_factory';
+import {createActivityInstance} from './activity_factory';
+
+export function parseUserTasks(processData: any): Array<Model.Activities.UserTask> {
+
+  const userTasks: Array<Model.Activities.UserTask> = [];
+
+  const userTasksRaw: Array<any> = getModelPropertyAsArray(processData, BpmnTags.TaskElement.UserTask);
+
+  if (!userTasksRaw || userTasksRaw.length === 0) {
```
Comment (Vyperus) [inline src/model/parser/flow_node_parsers/activity_parsers/user_task_parser.ts]: The extraction of this business rule is of uttermost importance for the readability of this code sir.

File: src/model/parser/flow_node_parsers/activity_parsers/user_task_parser.ts [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+      };
+      formField.enumValues = rawValues ? rawValues.map(valueMapper) : [];
+    }
+
+    return formField;
+  }
+
+  function parseDate(value: string): Date {
+
+    if (!value || value.length === 0 || !moment(value, 'YYYY-MM-DDTHH:mm:ss', true).isValid()) {
```
Comment (Vyperus) [inline src/model/parser/flow_node_parsers/activity_parsers/user_task_parser.ts]: So I guess this is checking if the value is a valid date?
If so:

```suggestion
    const isValidDate: boolean = !value || value.length === 0 || !moment(value, 'YYYY-MM-DDTHH:mm:ss', true).isValid();
    if (isValidDate) {
```

File: src/model/parser/process_parser.ts [hunk 3]
```diff
@@ -10,8 +10,11 @@ import {
   getModelPropertyAsArray,
 } from './../type_factory';
 
-import * as Parser from './index';
+import {Parsers} from './index';
 
+// TODO: The following elements are not supported yet:
+// - Text annotations
```
Comment (Vyperus) [inline src/model/parser/process_parser.ts]: Aren't text annotations perfectly fine? We use them for the heatmap feature or not?

File: src/runtime/flow_node_handler/empty_activity_handler.ts [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+  IFlowNodePersistenceFacade,
+  IProcessModelFacade,
+  IProcessTokenFacade,
+} from '@process-engine/process_engine_contracts';
+import {Model} from '@process-engine/process_model.contracts';
+
+import {FlowNodeHandlerInterruptible} from './index';
+
+// This type of handler doesn't actually do anything but pass on the token it receives.
+// Think of it as kind of a break point.
```
Comment (Vyperus) [inline src/runtime/flow_node_handler/empty_activity_handler.ts]:
```suggestion
// Think of it as a kind of break point.
```

File: src/model/parser/process_parser.ts (hunk 3 above)
Comment (ElRaptorus) [reply src/model/parser/process_parser.ts]: You use them in the Studio, yes, but we don't use them in the backend. 🙂 
Yet.

Comment (Vyperus) [reply src/model/parser/process_parser.ts]: Oh yeah nevermind.

**Pull request merged**